spirv_hlsl.cpp

```diff
@@ -1273,9 +1273,119 @@ void CompilerHLSL::emit_uniform(const SPIRVariable &var)
 {
 	add_resource_name(var.self);
 	auto &type = get<SPIRType>(var.basetype);
-	if (options.shader_model >= 40 && (type.basetype == SPIRType::Image || type.basetype == SPIRType::SampledImage))
+	if (options.shader_model >= 40 && type.basetype == SPIRType::SampledImage)
 	{
-		statement("Texture2D<float4> ", to_name(var.self), ";");
+		string format = "float4";
```

Comment:
Ah, that's much easier. Actual i/usamplers do not work though because a Texture2D<int4> can not be sampled - HLSL only supports the Load method in that case which is pixel position based and does no interpolation. Not yet sure how to handle that situation.